common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1480,7 +1499,7 @@ public void onComplete(String streamId) throws IOException {
           // Identify duplicate block generated by speculative tasks. We respond success to
           // the client in cases of duplicate even though no data is written.
           if (isDuplicateBlock()) {
-            freeDeferredBufs();
+            updateIgnoredBytes(freeDeferredBufs());
```

Comment:
```suggestion
            freeDeferredBufs();
            // Since we don't throw an exception and just return, but the received bytes are ignored we need to update the ignored bytes metric.
             receivedBytesIgnored();
```